src/test/java/edu/ksu/canvas/attendance/submitter/CanvasAssignmentAssistantUTest.java

```diff
@@ -0,0 +1,176 @@
+    private List<AttendanceSection> sectionList;
+
+    @Before
+    public void setup() {
+
+        canvasAssignmentAssistant = new CanvasAssignmentAssistant();
+
+        oauthToken = new NonRefreshableOauthToken(OAUTH_STRING);
+
+        attendanceAssignment = new AttendanceAssignment();
+        attendanceAssignment.setAssignmentId(ASSIGNMENT_ID);
+        attendanceAssignment.setAssignmentName(ASSIGNMENT_NAME);
+        attendanceAssignment.setCanvasAssignmentId(CANVAS_ASSIGNMENT_ID);
+        attendanceAssignment.setGradingOn(true);
+        attendanceAssignment.setAssignmentPoints(ASSIGNMENT_POINTS);
+        attendanceAssignment.setAbsentPoints(0.0);
+        attendanceAssignment.setExcusedPoints(0.0);
+        attendanceAssignment.setTardyPoints(0.0);
+        attendanceAssignment.setPresentPoints(100.0);
+
+        assignment = new Assignment();
+        assignment.setId(Long.toString(ASSIGNMENT_ID));
+        assignment.setName(attendanceAssignment.getAssignmentName());
+        assignment.setPointsPossible(Double.valueOf(attendanceAssignment.getAssignmentPoints()));
+        assignment.setCourseId(Long.toString(COURSE_ID));
+        assignment.setDescription(ASSIGNMENT_DESCRIPTION);
+        assignment.setPublished(true);
+        assignment.setUnpublishable(false);
+        assignmentOptional = Optional.of(assignment);
+
+        attendanceSection = new AttendanceSection();
+        attendanceSection.setCanvasCourseId(COURSE_ID);
+        attendanceSection.setSectionId(SECTION_ID);
+        attendanceSection.setName(SECTION_NAME);
+        attendanceSection.setCanvasSectionId(SECTION_ID);
+        sectionList = new ArrayList<>();
+        sectionList.add(attendanceSection);
+        attendanceAssignment.setAttendanceSection(attendanceSection);
+
+        Whitebox.setInternalState(canvasAssignmentAssistant, "assignmentService", assignmentService);
+        Whitebox.setInternalState(canvasAssignmentAssistant, "attendanceSectionService", attendanceSectionService);
+        Whitebox.setInternalState(canvasAssignmentAssistant, "assignmentRepository", assignmentRepository);
+        Whitebox.setInternalState(canvasAssignmentAssistant, "canvasApiWrapperService", canvasApiWrapperService);
+    }
+
+    @Test
+    public void createAssignmentInCanvasHappyPath() throws IOException, AttendanceAssignmentException {
+        when(canvasApiWrapperService.createAssignment(any(), any(), any())).thenReturn(assignmentOptional);
+        when(attendanceSectionService.getSectionsByCourse(COURSE_ID)).thenReturn(sectionList);
+        when(assignmentService.findBySection(any())).thenReturn(attendanceAssignment);
+        when(assignmentRepository.save(attendanceAssignment)).thenReturn(attendanceAssignment);
+
+        canvasAssignmentAssistant.createAssignmentInCanvas(COURSE_ID, attendanceAssignment, oauthToken);
+    }
+
+    @Test
+    public void createAssignmentInCanvasNullReturnError() throws Exception {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS00112)